src/Maestro/Maestro.DataProviders/ConfigurationIngestion/ConfigurationIngestor.cs

```diff
@@ -0,0 +1,388 @@
+            .Select(channel => SqlBarClient.ToClientModelChannel(channel))
+            .Select(ChannelYaml.FromClientModel)
+            .Select(yamlChannel => new IngestedChannel(yamlChannel))
+            .ToList();
+
+        var defaultChannelDaos = await _context.DefaultChannels
+            .Where(dc => dc.Namespace == namespaceEntity)
+            .ToListAsync();
+
+        var convertedDefaultChannels = defaultChannelDaos
+            .Select(dc => SqlBarClient.ToClientModelDefaultChannel(dc))
+            .Select(DefaultChannelYaml.FromClientModel)
+            .Select(yamlDc => new IngestedDefaultChannel(yamlDc))
+            .ToList();
+
+        var branchMergePolicieDaos = await _context.RepositoryBranches
+            .Where(rb => rb.Namespace == namespaceEntity)
+            .ToListAsync();
+
+        var convertedBranchMergePolicies = branchMergePolicieDaos
+            .Select(rb => SqlBarClient.ToClientModelRepositoryBranch(rb))
+            .Select(BranchMergePoliciesYaml.FromClientModel)
+            .Select(rbYaml => new IngestedBranchMergePolicies(rbYaml))
+            .ToList();
+
+        return new ConfigurationData(
+            convertedSubscriptions,
+            convertedChannels,
+            convertedDefaultChannels,
+            convertedBranchMergePolicies);
+    }
+
+    private async Task<Namespace> FetchOrCreateNamespace(string configurationNamespace)
+    {
+        var namespaceEntity = await _context.Namespaces
+            .FirstOrDefaultAsync(ns => ns.Name == configurationNamespace);
+
+        if (namespaceEntity is null)
+        {
+            namespaceEntity = new Namespace
+            {
+                Name = configurationNamespace,
+            };
+            _context.Namespaces.Add(namespaceEntity);
+            await _context.SaveChangesAsync();
+        }
+
+        return namespaceEntity;
+    }
+
+    private async Task CreateSubscriptions(
+        IEnumerable<IngestedSubscription> newSubscriptions,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName)
+    {
+        List<Subscription> subscriptionDaos = [.. newSubscriptions
+            .Select(sub => ConfigurationDataHelper.ConvertIngestedSubscriptionToDao(
+                sub,
+                namespaceEntity,
+                existingChannelsByName))];
+
+        await _sqlBarClient.CreateSubscriptionsAsync(subscriptionDaos, false);
+    }
+
+    private async Task UpdateSubscriptions(
+        IEnumerable<IngestedSubscription> updatedSubscriptions,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName)
+    {
+        List<Subscription> subscriptionDaos = [.. updatedSubscriptions
+            .Select(sub => ConfigurationDataHelper.ConvertIngestedSubscriptionToDao(
+                sub,
+                namespaceEntity,
+                existingChannelsByName))];
+
+        await _sqlBarClient.UpdateSubscriptionsAsync(subscriptionDaos, false);
```

Comment:
it's a bit weird that we only use the sqlBarClient for subscriptions. Is there any real reason to have it there?